charts/argo-workflows/templates/controller/workflow-controller-service.yaml

```diff
@@ -6,7 +6,7 @@ metadata:
   namespace: {{ .Release.Namespace | quote }}
   labels:
     {{- include "argo-workflows.labels" (dict "context" . "component" .Values.controller.name "name" .Values.controller.name) | nindent 4 }}
-    app.kubernetes.io/version: {{ default (include "argo-workflows.defaultTag" .) .Values.controller.image.tag | trunc 63 | quote }}
+    app.kubernetes.io/version: {{ include "argo-workflows.controller_chart_version_label" . }}
```

Comment:
📝  `app.kubernetes.io/version` showed correct value as expected with the following values.
```yaml
controller:
  serviceLabels:
    testing: "1234"
  image:
    registry: 1234.dkr.ecr.us-west-2.amazonaws.com
    repository: third-party/argoproj/workflow-controller
    tag: v3.5.0@sha256:dda36f7a872f463a68df04d72755c086eb193fa1645e1f099f1670dd5f99a277
  metricsConfig:
    enabled: true
server:
  image:
    registry: 1234.dkr.ecr.us-west-2.amazonaws.com
    repository: third-party/argoproj/argocli
    tag: v3.5.0@sha256:608b6d532b6a1ad4ae82b1e05ee844eb2f4fffe90a8524421f95e9355a7e4be1
  secure: true
  authMode: sso
```
<img width="831" alt="image" src="https://github.com/argoproj/argo-helm/assets/30188755/02f12d7e-51a4-4ce3-b9cb-d661b7c57542">